src/daemon/distribution.c

```diff
@@ -0,0 +1,148 @@
+#include <math.h>
+#include "collectd.h"
+#include "distribution.h"
+
+struct bucket_s{
+	size_t bucket_counter;
+	double min_boundary;
+	double max_boundary;
+};
+
+struct distribution_s {
+     bucket_t *buckets;
+     size_t num_buckets;
+     size_t total_scalar_count;
+     double raw_data_sum;
+};
+
+//bucket constructor
+bucket_t initialize_bucket(double min_boundary, double max_boundary) {
+    bucket_t new_bucket = {
+        .bucket_counter = 0,
+        .min_boundary = min_boundary,
+        .max_boundary = max_boundary,
+    };
+    return new_bucket;
+} 
+
+//increment bucket_counter of specific bucket
+void increment(bucket_t *bucket) { 
+    bucket->bucket_counter++;
+} 
+
+distribution_t * distribution_new_linear(size_t num_buckets, double size) {
+    distribution_t *new_distribution = calloc(1, sizeof(distribution_t));
```

Comment:
`calloc` may fail and return `NULL`. Please handle this case gracefully, i.e. without crashing due to a NULL pointer dereference.

Reply:
I am checking for that now.